File: 백준/Silver/13305. 주유소/주유소.py
n = int(input())
km = list(map(int, input().split()))
girum = list(map(int, input().split()))

total_biyong = girum[0] * km[0]

girum.pop(0)
girum.pop(-1)
girum.sort()
km.pop(0)
sum_km = sum(km)
total_biyong += girum[0] * sum_km
print(total_biyong)

File: 백준/Silver/13305. 주유소/주유소.js
let fs = require("fs");

let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let n = Number(input[0]);
let lengths = input[1].split(" ").map(Number);
let prices = input[2].split(" ").map(Number);
let newPrices = [];
let answer = 0;
let totalLength = 0;
for(let i = 0; i < prices.length - 1; i++){
  newPrices.push(prices[i]);
  totalLength += lengths[i];
}

newPrices.sort((a, b) => a - b);

for(let i = 0; i < newPrices.length; i++){
  if(newPrices[0] === prices[i]){
    answer += newPrices[0] * totalLength;
    break;
  }
  else{
    answer += prices[i] * lengths[i];
    totalLength -= lengths[i];
  }
}
console.log(answer);
